src/main/java/lavalink/client/io/mock/MockLavalink.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Written by TheMulti0, 2019
+ * https://github.com/TheMulti0
+ */
+
+package lavalink.client.io.mock;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import lavalink.client.io.Lavalink;
+
+import java.net.URI;
+
+public class MockLavalink extends Lavalink {
```

Comment:
It is my understanding both abstract classes can be mocked using Mockito to the same extent as these two new classes. If that is so, then the Mockito library should not be included as a main dependency.

Reply:
They cannot. I wanted to mock them manually so I can still have the custom parameterized constructor for the MockLink, and the MockLavalink to use MockLink.
Also the MockLavalink does not have a parameterized constructor, unlike its abstract base class.

Reply:
Have you tried? The type parameter doesn't seem to be problematic.

![image](https://user-images.githubusercontent.com/2582617/57853719-c0107c00-77e6-11e9-973c-20aedb67e76b.png)

Reply:
Yep, threw an exception for me, didn't really look into it just because I assumed it wont work.
Did you test if it's actually working without throwing an exception / if playing correctly?

Reply:
It works. I added a comment showing the output from stdout.